test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -447,6 +449,407 @@ def test_ir_non_empty_instructions_result_types_basis_rotation_instructions():
     assert circ.to_ir() == expected
 
 
+@pytest.mark.xfail(raises=ValueError)
+def test_as_unitary_empty_instructions_raises_error():
+    circ = Circuit()
+    circ.as_unitary() == []
+
+
+@pytest.mark.parametrize("targets", [(10), (11), (12)])
+def test_as_unitary_10_qubits_circuit_produce_warning(targets):
+    logger = Mock()
+    Circuit(logger=logger).i(range(targets)).as_unitary()
+
+    logger.warning.assert_called_once_with(
+        f"To produce an unitary for 10 or more qubits can get slow. Current count: {targets}"
+    )
+
+
+@pytest.mark.parametrize(
+    "circuit",
+    [
+        (Circuit().phaseshift(0, 0.15).apply_gate_noise(noise.Noise.BitFlip(probability=0.1))),
+        (Circuit().cnot(1, 0).apply_gate_noise(noise.Noise.TwoQubitDepolarizing(probability=0.1))),
+        (
+            Circuit()
+            .x(1)
+            .i(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[1])
+        ),
+        (
+            Circuit()
+            .x(1)
+            .i(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[2])
+        ),
+        (Circuit().x(1).i(2).apply_gate_noise(noise.Noise.BitFlip(probability=0.1))),
+        (Circuit().x(1).apply_gate_noise(noise.Noise.BitFlip(probability=0.1)).i(2)),
+        (
+            Circuit()
+            .y(1)
+            .z(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[1])
+        ),
+        (
+            Circuit()
+            .y(1)
+            .z(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[2])
+        ),
+        (Circuit().y(1).z(2).apply_gate_noise(noise.Noise.BitFlip(probability=0.1))),
+        (Circuit().y(1).apply_gate_noise(noise.Noise.BitFlip(probability=0.1)).z(2)),
+        (
+            Circuit()
+            .cphaseshift(2, 1, 0.15)
+            .si(3)
+            .apply_gate_noise(
+                noise.Noise.TwoQubitDepolarizing(probability=0.1), target_qubits=[1, 2]
+            )
+        ),
+        (
+            Circuit()
+            .cphaseshift(2, 1, 0.15)
+            .apply_gate_noise(noise.Noise.TwoQubitDepolarizing(probability=0.1))
+            .si(3)
+        ),
+    ],
+)
+@pytest.mark.xfail(raises=TypeError)
+def test_as_unitary_noise_raises_error(circuit):
+    circuit.as_unitary()
+
+
+def test_as_unitary_noise_not_apply_returns_expected_unitary(recwarn):
```

Comment:
This test was writing in a different way because the `warning` is launched as soon as we add noise, so we had to instantiate `recwarn` before creating the circuit.

Reply:
This might be due to an issue with the way qubits are indexed. Could you try adding a `.i(0)` to the circuit, that is:
```
circuit = (
        Circuit()
        .i(0)
        .cphaseshift(2, 1, 0.15)
        .si(3)
        .apply_gate_noise(noise.Noise.TwoQubitDepolarizing(probability=0.1), target_qubits=[1, 3])
    )
```

Reply:
I tried but the warning is still printed as soon as the noise gate is applied. As far as I can see, the warning is printed because the target qubits in the noise gate does not match the target qubits in the only other 2-qubit gate in the circuit (this is how I understand this code: https://github.com/aws/amazon-braket-sdk-python/blob/main/src/braket/circuits/noise_helpers.py#L209, although I might be wrong).

Anyway, this test might be unnecessary, I just wanted to make clear that a noise gate will trigger an error only if the noise gate is in fact applied. If the noise gate is ignored, there is nothing I can do about it.